src/reqstool/requirements_indata/requirements_indata_paths.py

```diff
@@ -40,12 +41,23 @@ def prepend_paths(self, prepend_dir):
         for prop in dir(self):
             if (
                 not prop.startswith("__")
-                and isinstance(getattr(self, prop), RequirementsIndataPathItem)
+                and self.attribute_is_instance_or_list_of_class(prop_name=prop, the_class=RequirementsIndataPathItem)
                 and prop not in ["requirements_yml", "svcs_yml", "mvrs_yml"]
             ):
                 path_item = getattr(self, prop)
-                if path_item.path is not None:
-                    path_item.path = str(PurePath(prepend_dir, path_item.path))
+                self.check_type_and_prepend_path(path_item=path_item, prepend_dir=prepend_dir)
+
+    # prepend path for each entry in a list of RequirementsIndataPathItem(s) or just on a single entity of the class
+    def check_type_and_prepend_path(self, path_item, prepend_dir: str):
+        if type(path_item) is list:
+            for entry in path_item:
+                if type(entry) is RequirementsIndataPathItem and entry.path is not None:
```

Comment:
Can this happen? We are parsing into RequirementsIndataPathItem right? I don't think we shall just log and continue. There is a serious issue with the parsing if this fails.

Reply:
```python
def check_type_and_prepend_path(self, path_item, prepend_dir: str):
    if isinstance(path_item, list):
        if all(isinstance(entry, RequirementsIndataPathItem) and entry.path is not None for entry in path_item):
            path_item = [entry.path = str(PurePath(prepend_dir, entry.path)) for entry in path_item]
    else:
        if path_item.path is not None:
            path_item.path = str(PurePath(prepend_dir, path_item.path))

```

Reply:
This is really just a case for the test_result_dirs as they are expressed as a list of RequirementIndataPathItems

For example, the MavenDefault Paths

![image](https://github.com/Luftfartsverket/reqstool-client/assets/142776362/bd07bfcb-5226-4a11-bf8a-05fa3e9d45f9)


without this logic, the paths for those objects are not handled correctly.

Reply:
However, I agree that just a log in not robust enough to handle this. I could throw an exception as well.